datetime/src/commonMain/kotlin/datecomponents/DateComponents.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.datetime.datecomponents
+
+import com.splendo.kaluga.datetime.DateTime
+import com.splendo.kaluga.datetime.DateTimeComponents
+import com.splendo.kaluga.datetime.components
+import com.splendo.kaluga.datetime.dayDateComponent
+import com.splendo.kaluga.datetime.gmtDateTime
+import com.splendo.kaluga.datetime.monthDateComponent
+import com.splendo.kaluga.datetime.yearDateComponent
+
+sealed class DateComponents {
+    abstract fun dateTime(): DateTime?
+
+    companion object {
+        fun undefinedYear(): DateComponentsUndefined =
+            DateComponentsUndefined(
+                Year(0)
+            )
+
+        fun undefinedMonth(): DateComponentsUndefined =
+            DateComponentsUndefined(
+                Month(0, 0)
+            )
+
+        fun undefinedDay(): DateComponentsUndefined =
+            DateComponentsUndefined(
+                Day(0, 0, 0)
+            )
+    }
+
+    fun intYear(): Int? =
```

Comment:
Why not make this `abstract val intYear: Int?` and implement it in the class itself

Reply:
Yes. Right. It was implemented as an extension, but it is not exposing to swift and I moved it to class self and now it makes sense to use abstract.